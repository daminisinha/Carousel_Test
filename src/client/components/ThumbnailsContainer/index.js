import React, { Component } from 'react';
import { Row, Col, Visible } from 'react-grid-system';
import Slide from '../Slider'
import './index.scss'
class Thumbnails extends Component {
    render() {
        let { images,onClickHandler } = this.props;
        let slides = [];
        for(let i=0;i<images.length;i++){
            let img = images[i]
            slides.push(
                <li className="thumbnails-item" key={img.id}><Slide 
                          onClickHandler={onClickHandler}
                          index={i}
                          image={img}
                          width={img.previewWidth}
                          height="100" />
                    <h5>By: {img.user}</h5>
                </li> 
            )
        }
        
        return (
            <Row className="thumbnails">
                <Col  lg={12} xl={12}>
                     
                    <Visible lg xl>
                        <ul className="thumbnails-list">
                             {slides}
                       </ul>
                    </Visible>

                </Col>
            </Row>
        );
    }
}

export default Thumbnails;