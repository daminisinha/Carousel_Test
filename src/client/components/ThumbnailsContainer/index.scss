.thumbnails{
    text-align:center;
}

.thumbnails-list{
    display:flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: space-evenly;
    list-style-type: none;
    margin-bottom:0px;
}

.thumbnails-item{
    cursor:pointer;
}

