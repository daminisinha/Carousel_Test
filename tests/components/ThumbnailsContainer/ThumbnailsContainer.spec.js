import React from 'react';
import ThumbnailsContainer from '.../../../src/client/components/ThumbnailsContainer'
import {shallow} from 'enzyme';

test('Thumbnail renders',()=>{
    let images = [1,2,3,4,5,6]
    const component = shallow(<ThumbnailsContainer images={images}/>); 
    expect(component).toMatchSnapshot();
})
