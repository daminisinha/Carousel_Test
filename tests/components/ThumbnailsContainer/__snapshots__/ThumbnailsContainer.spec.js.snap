// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders 1`] = `
<Row
  align="normal"
  className="thumbnails_row"
  component="div"
  debug={false}
  gutterWidth={null}
  justify="start"
  nogutter={false}
  nowrap={false}
  style={Object {}}
>
  <Col
    component="div"
    debug={false}
    lg={12}
    md={null}
    offset={Object {}}
    pull={Object {}}
    push={Object {}}
    sm={null}
    style={Object {}}
    xl={12}
    xs={null}
  >
    <Visible
      lg={true}
      md={false}
      sm={false}
      xl={true}
      xs={false}
    >
      <ul
        className="thumbnails__list"
      >
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={1}
            index={0}
          />
          <h5>
            Author: 
          </h5>
        </li>
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={2}
            index={1}
          />
          <h5>
            Author: 
          </h5>
        </li>
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={3}
            index={2}
          />
          <h5>
            Author: 
          </h5>
        </li>
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={4}
            index={3}
          />
          <h5>
            Author: 
          </h5>
        </li>
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={5}
            index={4}
          />
          <h5>
            Author: 
          </h5>
        </li>
        <li
          className="thumbnails__item"
        >
          <Slider
            height="100"
            image={6}
            index={5}
          />
          <h5>
            Author: 
          </h5>
        </li>
      </ul>
    </Visible>
  </Col>
</Row>
`;
